Automattic/woocommerce-payments
# Store Stripe exchange rate after a purchase

Fixes #2567 

#### Changes proposed in this Pull Request

Store Stripe exchange rate after a purchase

- A new method is added inside of WC Payments Gateway to store the Stripe exchange rate under `_wcpay_multi_currency_stripe_exchange_rate` in each order.
- The rate is obtained using the Charge info (I tried to avoid making a new API request for this but I couldn't find the rate in any other place).
- Before making the request I'm checking if it's needed comparing the WC Payments currency with the currency of the order. If it's the same, the exchange rate doesn't appear in the Charge info.
- This stored rate will be used in the Analytics.

#### Testing instructions

* Make a new purchase using a currency that is not the same as the one set in the WC Payments account.
* Use phpMyAdmin to see in the metadata of the order the exchange rate stored (not sure if there is a more efficient way to check this in the wp-admin!)

![image](https://user-images.githubusercontent.com/2612426/127296355-9de35cef-adc6-41bb-9529-e090469309a7.png)


-------------------
